engine/src/main/java/org/hibernate/search/engine/service/impl/StandardServiceManager.java

```diff
@@ -108,10 +113,24 @@ public StandardServiceManager(SearchConfiguration searchConfiguration,
 
 	@Override
 	public void releaseAllServices() {
-		for ( ServiceWrapper wrapper : cachedServices.values() ) {
-			wrapper.ensureStopped();
+		List<String> unreleasedServicesToReport = failOnUnreleasedService ? new ArrayList<String>() : null;
+
+		for ( ServiceWrapper<?> wrapper : cachedServices.values() ) {
+			boolean stoppedProperly = wrapper.ensureStopped();
```

Comment:
the wrapper needs to be synchronized here, like the original method which you're inlining now.
Failing to do so might expose you to a service instance to increment its usage counter during this stop operation for a legitimate usage, and yet throw the exception (or log the warning).

It's extremely unlikely as we'd be shutting down anyway.. but it's in theory possible if e.g. a service is needed to perform a clean teardown.
I'll fix that.